src/js/components/FileInput/FileInput.js

```diff
@@ -214,49 +259,76 @@ const FileInput = forwardRef(
+                        inputRef.current.click();
+                    }}
+                    onEnter={event => {
+                      if (controlRef.current === event.target)
+                        inputRef.current.click();
+                    }}
+                  >
+                    {theme.fileInput.button ? (
+                      <Button
+                        ref={controlRef}
+                        kind={theme.fileInput.button}
+                        label={messages.browse || 'browse'}
+                        onClick={() => {
+                          inputRef.current.click();
+                          inputRef.current.focus();
+                        }}
+                      />
+                    ) : (
+                      <Anchor
+                        tabIndex={0}
+                        ref={controlRef}
+                        margin="small"
+                        onClick={() => {
+                          inputRef.current.click();
+                          inputRef.current.focus();
+                        }}
+                        label={messages.browse || 'browse'}
+                      />
+                    )}
+                  </Keyboard>
+                )}
               </Box>
-            ))}
-          {!files.length && (
-            <Message {...theme.fileInput.message}>
-              {multiple
-                ? messages.dropPromptMultiple || 'Drop files here or'
-                : messages.dropPrompt || 'Drop file here or'}{' '}
-              <Anchor label={messages.browse || 'browse'} />
-            </Message>
-          )}
-          <StyledFileInput
-            ref={inputRef}
-            type="file"
-            id={id}
-            name={name}
-            multiple={multiple}
-            disabled={disabled}
-            plain
-            rightOffset={rightOffset}
-            {...rest}
-            onDragOver={() => setDragOver(true)}
-            onDragLeave={() => setDragOver(false)}
-            onChange={event => {
-              event.persist();
-              const fileList = event.target.files;
-              const nextFiles = multiple ? [...files] : [];
-              for (let i = 0; i < fileList.length; i += 1) {
-                // avoid duplicates
-                const existing =
-                  nextFiles.filter(
-                    file =>
-                      file.name === fileList[i].name &&
-                      file.size === fileList[i].size,
-                  ).length > 0;
-                if (!existing) nextFiles.push(fileList[i]);
-              }
-              setFiles(nextFiles);
-              setDragOver(false);
-              if (onChange) onChange(event);
-            }}
-          />
-        </ContentsBox>
-      </Keyboard>
+            </Box>
+          ))}
+        <StyledFileInput
+          ref={inputRef}
+          type="file"
+          id={id}
+          name={name}
+          multiple={multiple}
```

Comment:
Something got weird when I dragged in multiple files:
<img width="432" alt="Screen Shot 2021-05-11 at 4 30 00 PM" src="https://user-images.githubusercontent.com/12522275/117896816-3cbf7f00-b276-11eb-9dbc-ed0815c8826f.png">

Reply:
aw yes I added the `fill` prop and now should be good since I have it set to `horizontal` for the `label` and `anchor` to fill the `box`

Reply:
Nice, looks like it's fixed now.